### PR TITLE
feat: allow specification of replicas in env

### LINKED ISSUE #150: feat(configuration): allow custom configuration through env vars
- We allow agents to take custom config files
- Want to be able to pass in replicas through env for custom configuration

### PR DESCRIPTION
**Pull request by ltchang2019**
## Motivation

Easiest way for an external party to run agents is to only specify the remote networks it is interested in and drawing from our own configs. Easier than providing an entirely new config file.

## Solution

Allow operator to specify list of replicas as env vars. This also deprecates the ability to provide a `secrets.json` file, as it no longer aligns with the pattern of specifying remote networks. The feature is also unused at the moment.

```
AGENT_REPLICA_0=ethereum
AGENT_REPLICA_1=moonbeam
etc...

or just

AGENT_REPLICAS_ALL=true
```

## PR Checklist

- [x] Added Tests
- [x] Updated Documentation
- [x] Updated CHANGELOG.md for the appropriate package

Closes #150

### CODE REVIEW

File: fixtures/env.test [hunk 1]
```diff
@@ -2,16 +2,22 @@
 
 RUN_ENV=test
 AGENT_HOME=ethereum
+AGENT_REPLICAS_ALL=true
```
Comment (arnaud036) [inline fixtures/env.test:5]: Is this env var needed? Default behavior can be if no AGENT_REPLICA_{} is specified then we use all replicas defined in the config? So we avoid wrong combinations such as AGENT_REPLICAS_ALL=true and AGENT_REPLICA_0=luckspecialnetwork

File: fixtures/env.external [hunk 2]
```diff
@@ -1,21 +1,17 @@
 # Matches fixtures/external_config.json
 
 AGENT_HOME=ethereum
+AGENT_REPLICA_0=polygon
```
Comment (arnaud036) [inline fixtures/env.external:4]: Can we have this var suffixed with the expected content. AGENT_REPLICA_0_NAME?

File: fixtures/env.test (hunk 1 above)
Comment (ltchang2019) [reply fixtures/env.test:5]: Default is that if `AGENT_REPLICAS_ALL` is unspecified, code will look for `AGENT_REPLICA_{}`. So not a required field

File: fixtures/env.external (hunk 2 above)
Comment (ltchang2019) [reply fixtures/env.external:4]: If we do it should also be `AGENT_HOME_NAME` and `AGENT_REPLICA_{}_NAME`. Honestly think this is fine as is but up to you

Comment (arnaud036) [reply fixtures/env.external:4]: If it's easy to change my personal preference would be AGENT_HOME_NAME, if not I am ok to keep it that way.

Comment (ltchang2019) [reply fixtures/env.external]: k changed

File: nomad-base/src/settings/macros.rs [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+    ($home:ident, $config:ident) => {{
+        let connections = $config
+            .protocol()
+            .networks
+            .get(&$home)
+            .expect("!networks")
+            .connections
+            .clone();
+
+        if let Ok(_) = std::env::var("AGENT_REPLICAS_ALL") {
```
Comment (prestwich) [inline nomad-base/src/settings/macros.rs:13]: Has @arnaud036 reviewed these new env vars?

**Review by arnaud036 — APPROVED** [review]

File: nomad-base/src/settings/macros.rs (hunk 3 above)
Comment (arnaud036) [inline nomad-base/src/settings/macros.rs:13]: Even if AGENT_REPLICAS_ALL=false, agents behaves like AGENT_REPLICAS_ALL = true.

File: agents/kathy/src/settings.rs [hunk 4]
```diff
@@ -17,18 +17,24 @@ mod test {
...
             let run_env = dotenv::var("RUN_ENV").unwrap();
-            let agent_home = dotenv::var("AGENT_HOME").unwrap();
+            let agent_home = dotenv::var("AGENT_HOME_NAME").unwrap();
 
             let settings = KathySettings::new().unwrap();
 
             let config = nomad_xyz_configuration::get_builtin(&run_env).unwrap();
-            let secrets = AgentSecrets::from_env(&config.networks).unwrap();
+
+            let remotes = get_remotes_from_env!(agent_home, config);
```
Comment (arnaud036) [inline agents/kathy/src/settings.rs:26]: Can you add logs to show which network are being loaded?

File: nomad-base/src/settings/macros.rs (hunk 3 above)
Comment (ltchang2019) [reply nomad-base/src/settings/macros.rs]: Fixed

File: agents/kathy/src/settings.rs (hunk 4 above)
Comment (ltchang2019) [reply agents/kathy/src/settings.rs:26]: Added log for all remotes loaded

**Pull request merged**